testing/test_pluginmanager.py

```diff
@@ -78,6 +78,21 @@ def pytest_addoption(parser):
         )
         assert config.option.test123
 
+    def test_do_option_postinit_nodefault(self, pytester: Pytester) -> None:
+        config = pytester.parseconfigure()
+        assert not hasattr(config.option, "deadbeef")
+        p = pytester.makepyfile(
+            """
+            def pytest_addoption(parser):
+                parser.addoption('--deadbeef', action="store_true")
+        """
+        )
+        with pytest.raises(ValueError):
```

Comment:
just to be sure we should match the exception
we might want to add a full test for it

Reply:
adding the match for that exception and also edited one of the other tests which will catch the warning I'm putting out in cases where projects do use this with the default option. 

I realized that this is probably a very rare occurrence because adding options after config and parse phase will not make them available at the command line and will just crash pytest. I can provide a test app showing this if it better helps get my reasoning across.